plugins/org.eclipse.n4js/src/org/eclipse/n4js/typesystem/N4IDLVersionResolver.xtend

```diff
@@ -0,0 +1,157 @@
+/**
+ * Copyright (c) 2016 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.typesystem
+
+import com.google.inject.Inject
+import java.math.BigDecimal
+import java.util.List
+import java.util.ListIterator
+import org.eclipse.n4js.n4JS.VersionedElement
+import org.eclipse.n4js.n4idl.versioning.VersionHelper
+import org.eclipse.n4js.ts.typeRefs.ComposedTypeRef
+import org.eclipse.n4js.ts.typeRefs.FunctionTypeExpression
+import org.eclipse.n4js.ts.typeRefs.FunctionTypeRef
+import org.eclipse.n4js.ts.typeRefs.ParameterizedTypeRef
+import org.eclipse.n4js.ts.typeRefs.ParameterizedTypeRefStructural
+import org.eclipse.n4js.ts.typeRefs.TypeArgument
+import org.eclipse.n4js.ts.typeRefs.TypeRefsPackage
+import org.eclipse.n4js.ts.typeRefs.TypeTypeRef
+import org.eclipse.n4js.ts.typeRefs.VersionedFunctionTypeRef
+import org.eclipse.n4js.ts.typeRefs.VersionedParameterizedTypeRef
+import org.eclipse.n4js.ts.typeRefs.VersionedParameterizedTypeRefStructural
+import org.eclipse.n4js.ts.typeRefs.VersionedReference
+import org.eclipse.n4js.ts.types.ContainerType
+import org.eclipse.n4js.ts.types.TClassifier
+import org.eclipse.n4js.ts.types.TFormalParameter
+import org.eclipse.n4js.ts.types.TFunction
+import org.eclipse.n4js.ts.types.TMethod
+import org.eclipse.n4js.ts.types.Type
+import org.eclipse.n4js.ts.utils.TypeUtils
+import org.eclipse.n4js.typesystem.VersionResolver
+
+/**
+ * An implementation of {@link VersionResolver} that performs actual resolving of versions against the given context
+ * version. Generally, when resolving an actual version of a versioned type, this implementation will delegate to
+ * {@link VersionHelper#findClassifierWithVersion(TClassifier, int)}.
+ */
+class N4IDLVersionResolver implements VersionResolver {
```

Comment:
maybe we should now move this class and the implemented interface to package `org.eclipse.n4js.n4idl.versioning`. What do you think?